helix-core/src/main/java/org/apache/helix/task/WorkflowDispatcher.java

```diff
@@ -165,7 +165,9 @@ public void updateWorkflowStatus(String workflow, WorkflowConfig workflowCfg,
     RuntimeJobDag runtimeJobDag = _clusterDataCache.getTaskDataCache().getRuntimeJobDag(workflow);
     if (runtimeJobDag != null) {
       for (String inflightJob : runtimeJobDag.getInflightJobList()) {
-        processJob(inflightJob, currentStateOutput, bestPossibleOutput, workflowCtx);
+        if (System.currentTimeMillis() >= workflowCtx.getJobStartTime(inflightJob)) {
```

Comment:
For the execution delay itself, adding this check may do the trick. However, two more things to consider:
1. Would this be the right place to place the check? Say, should we include this check inside processJob()? Or should we create a separate method that checks whether a job is eligible to be processed?
2. Could there be more checks we need to perform other than execution delay? It would be great if we could make this a comprehensive fix :)

Reply:
Thank you for your comments.
1- Based on same logic that is added few line below (which calculates the start time and store these information and process the job if possible), this is the place that this check should happen. (same as the scheduleJobs method in WorkflowDistpatcher).
2- About the comprehensive change, I agree with you. This is just a quick fix. We will need to revisit several concepts such as inflight jobs and retry delay. I am working on the retry delay and will create new issue for that as well. After that we can think on a comprehensive changes to the Task Framework.